src/AzureStorageSagaExporter/CloudTableExtensions.cs

```diff
@@ -19,13 +19,13 @@ public static async IAsyncEnumerable<T> ExecuteQueryAsync<T>(this CloudTable tab
                         token: token,
                         requestOptions: null,
                         operationContext: null,
-                        cancellationToken: ct)
+                        cancellationToken: cancellationToken)
                     .ConfigureAwait(false);
                 token = seg.ContinuationToken;
 
                 foreach (var entity in seg.Results)
                 {
-                    if (alreadyTaken < take && !ct.IsCancellationRequested)
+                    if (alreadyTaken < take && !cancellationToken.IsCancellationRequested)
```

Comment:
I'm no expert on `IAsyncEnumerable` nor `EnumeratorCancellation` but something looks off to me here. The method won't throw on cancellation.